src/main/java/org/broadinstitute/hellbender/tools/spark/utils/LargeLongHopscotchSet.java

```diff
@@ -0,0 +1,224 @@
+
+import com.esotericsoftware.kryo.DefaultSerializer;
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import org.broadinstitute.hellbender.tools.spark.sv.SVUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+/**
+ * Set of longs that is larger than the max Java array size ( ~ 2^31 ~ 2 billion) and therefore cannot fit into a
+ * single LongHopscotchSet. Maintains partitions of LongHopscotchSets given a maximum partition size and the number
+ * of elements to be added. Bins each entry into a corresponding LongHopscotchSet using the entry hash value. Does
+ * not support dynamic resizing.
+ */
+@DefaultSerializer(LargeLongHopscotchSet.Serializer.class)
+public class LargeLongHopscotchSet implements Serializable, QueryableLongSet {
+
+    private static final long serialVersionUID = 1L;
+    private final List<LongHopscotchSet> sets;
+    private int numSets;
+
+    public LargeLongHopscotchSet(final long maxPartitionBytes, final long numElements) {
+        sets = new ArrayList<>();
+        initialize(maxPartitionBytes, numElements);
+    }
+
+    @SuppressWarnings("unchecked")
+    protected LargeLongHopscotchSet(final Kryo kryo, final Input stream) {
+        final boolean oldReferences = kryo.getReferences();
+        kryo.setReferences(false);
+
+        numSets = stream.readInt();
+        sets = new ArrayList<>(numSets);
+        for (int i = 0; i < numSets; i++) {
+            sets.add(kryo.readObject(stream, LongHopscotchSet.class));
+        }
+
+        kryo.setReferences(oldReferences);
+    }
+
+    private static int longHash(final long entryVal) {
+        return (int) SVUtils.fnvLong64(0x4c79ac6a, entryVal);
+    }
+
+    private void initialize(final long maxPartitionBytes, final long numElements) {
+        Utils.validateArg(maxPartitionBytes > 0, "Max partition size must be greater than 0");
+        Utils.validateArg(numElements > 0, "Number of elements must be greater than 0");
+
+        //Largest partition according to the max bytes per partition constraint
+        final int maxPartitionSize = LongHopscotchSet.getLegalSizeBelow(maxPartitionBytes / LongHopscotchSet.bytesPerBucket);
+        //Truncate numElements to largest int possible
+        final int idealPartitionSize = LongHopscotchSet.getPartitionSize(numElements);
+        //Number of elements per partition
+        final int elementsPerPartitionGuess = Math.min(idealPartitionSize, maxPartitionSize);
+        final int partitions = (int) Math.ceil(((double) numElements) / elementsPerPartitionGuess);
+        final int elementsPerPartition = (int)Math.ceil(numElements / (double)partitions);
+
+        for (int i = 0; i < partitions; i++) {
+            sets.add(new LongHopscotchSet(elementsPerPartition));
```

Comment:
I'm not sure, but I think you may not have taken into account that elementsPerPartition will be inflated by LOAD_FACTOR and getLegalSizeAbove.